extras/rebalance/directory-rebalance.py

```diff
@@ -77,35 +88,45 @@ def init_logging(self):
 
     #Executes the setxattr syscall to trigger migration
     def migrate_data(self, f):
+        size_now = 0
         try:
+            size_now = os.stat(f).st_size
             os.setxattr(f, "trusted.distribute.migrate-data", b"1",
                         follow_symlinks=False)
-            return True, None #Indicate that migration happened
+            return True, size_now, None #Indicate that migration happened
 
         except OSError as e:
-            return False, e
+            return False, size_now, e
```

Comment:
I'm not sure if this is good. If the error happens at `os.stat()`, `size_now` will be 0, however if the error happens at `os.setxattr()`, `size_now` will have some value. You don't seem to be correctly handling these two cases in the caller function.

Reply:
I think I did. size_diff is updated to -size in case the errno is ENOENT. Did you find any other issue?

Agree about the microseconds thing. Will change it.

Reply:
Suppose the file is deleted before `os.stat()`. In this case this function will return 0. `size_diff` in `migrate_with_stats()` will be initialized as:

    size_diff = size_now - size = 0 - size = -size

Then you check the error, which will be ENOENT. Here you update `size_diff` as:

    size_diff = size_diff - size = -size - size = -2*size

So you are subtracting twice the real size of the file from the `expected_total_size` in line 123.

Reply:
Sorry, ignore my last comment... Let me check again...

Reply:
I misread line 115. I "saw" `size_diff` was updated by subtracting `size` instead of assigned... sorry... I will need to pay more attention...

I think it's fine now.

Reply:
Phew! Thought there was one more bug :-).